has-matching-release-tag/hasMatchingReleaseTag.ts

```diff
@@ -0,0 +1,77 @@
+import { exec, ExecOptions } from '@actions/exec'
+
+export async function hasMatchingReleaseTag(
+	refName: string,
+	releaseTagRegexp: RegExp,
+	releaseBranchWithoutPatchRegexp: RegExp,
+	releaseBranchWithPatchRegexp: RegExp,
+): Promise<string> {
+	let refNames: Array<string> = []
+	const options: ExecOptions = {
+		listeners: {
+			stdout: (data: Buffer) => {
+				refNames = data.toString().split(/(?:\r\n|\r|\n)/g)
+			},
+		},
+	}
+
+	await exec('git', ['tag'], options)
+
+	return hasMatchingReleaseTagWithRefNames(
+		refNames,
+		refName,
+		releaseTagRegexp,
+		releaseBranchWithoutPatchRegexp,
+		releaseBranchWithPatchRegexp,
+	)
+}
+
+export function filterRefNames(refNames: Array<string>, regexp: RegExp): Array<string> {
+	return refNames.filter((name) => name.match(regexp))
+}
+
+export function hasMatchingReleaseTagWithRefNames(
+	refNames: Array<string>,
+	refName: string,
+	releaseTagRegexp: RegExp,
+	releaseBranchWithoutPatchRegexp: RegExp,
+	releaseBranchWithPatchRegexp: RegExp,
+): string {
+	if (refName.match(releaseTagRegexp)) {
+		return 'true'
+	}
+
+	let releaseTags = filterRefNames(refNames, releaseTagRegexp)
+
+	let branchMatches = refName.match(releaseBranchWithPatchRegexp)
```

Comment:
As discussed, this use case is expected to be fairly uncommon.  I think we could make this optional by skipping if `releaseBranchWithPatchRegexp` is an empty string?   If so, it might be sensible to make that the default input value as well, and workflows wouldn't have to define it.

Reply:
Made optional in a92ad8b